odl/test/solvers/functional/default_functionals_test.py

```diff
@@ -614,5 +614,73 @@ def test_weighted_proximal_L1_norm_close(space):
     assert all_almost_equal(expected_result, p_ip)
 
 
+def test_bregman_functional_no_gradient(space):
+    """Test Bregman distance for functional without gradient.
+
+    Test that the Bregman distance functional fails if the underlying
+    functional does not have a gradient and no subgradient operator is
+    given. Also test giving the subgradient operator separately.
+    """
+
+    ind_func = odl.solvers.IndicatorNonnegativity(space)
+    point = noise_element(space)
+
+    # Indicator function has no gradient, hence one cannot create a bregman
```

Comment:
Frankly we should just add gradient=0 for the indicator function. It makes sense, no?

This would be for later though

Reply:
The indicator function on a set has gradient=0 on that set. But outside the set, where it takes value `inf`, I think that the subgradient is empty. So we could return `EmptySet` in these cases, but I don't know if that would make anyone happier or if it would just confuse people...?

Reply:
I think what you're talking about @aringh is the sub*differential*, not sub*gradient*. The subgradient is not a set, it's an (arbitrary) element from the subdifferential. So rather than returning an empty set, the `subgrad` operator should in principle fail if we're outside the set of the indicator function, since logically, we're trying to return an element from the empty set.

I don't know if that trouble is worth going through. Is there a case where we would need this?

Reply:
> Frankly we should just add gradient=0 for the indicator function. It makes sense, no?

I think this is mathematically wrong since 0 is not a subgradient in points outside the set of the indicator function. So IMO we shouldn't do this.